autonomy/deploy/base.py

```diff
@@ -69,33 +68,33 @@ def __init__(
         keys: Path,
         number_of_agents: Optional[int] = None,
         private_keys_password: Optional[str] = None,
+        agent_instances: Optional[List[str]] = None,
```

Comment:
we set information that is partially redundant here. Why do we even pass the number_of_agents as a parameter in the first place? Seems like we don't have to. At least I suppose we should check that the size of agent_instances matches the specified number_of_agents. But is there ever a situation where we pass only `number_of_agents` and not `agent_intances`? I think we may want to remove `number_of_agents`

https://github.com/valory-xyz/open-autonomy/blob/693915c76e8fe40907a66223687d2f3ebd6228da/autonomy/deploy/base.py#L65-L72

Reply:
We create two types of deployments, on-chain deployments and service deployments. Now with on-chain deployments your argument makes sense, but with the local deployments while developing service and components a user might want to experiment with a different number of agents without updating the service config or the keys file so we provide this option here.

Reply:
great. Lets add that in a sentence to the docstring or as a comment: "we allow configurable number of agents independent of the number of agent instances for local development purposes"

Reply:
@angrybayblade please add